server/src/test/java/io/crate/metadata/upgrade/MetadataIndexUpgraderTest.java

```diff
@@ -120,4 +124,21 @@ private static CompressedXContent createDynamicStringMappingTemplate() throws IO
 
         return new CompressedXContent(BytesReference.bytes(builder));
     }
+
+    @Test
+    public void test_copy_to_migrated_to_sources() throws Throwable {
+        IndexMetadata indexMetadata = IndexMetadata.builder(new RelationName("doc", "users").indexNameOrAlias())
+            .settings(Settings.builder().put("index.version.created", Version.V_5_3_0))
+            .numberOfShards(1)
+            .numberOfReplicas(0)
+            .putMapping(MappingConstants.FULLTEXT_MAPPING_5_3)
+            .build();
+
+        MetadataIndexUpgrader metadataIndexUpgrader = new MetadataIndexUpgrader();
+        IndexMetadata updatedMetadata = metadataIndexUpgrader.apply(indexMetadata, null);
+
+        MappingMetadata mapping = updatedMetadata.mapping();
+        assertThat(mapping.source().string())
+            .isEqualTo(MappingConstants.FULLTEXT_MAPPING_EXPECTED_IN_5_4);
+    }
```

Comment:
Given that this is quite a big change maybe wort adding an integration test with data folder snapshot taken on 5.3?